test/specs/parsing/evaluateXPath.tests.ts

```diff
@@ -83,6 +93,90 @@ describe('evaluateXPath', () => {
 	it('Requires the XPath selector', () =>
 		chai.assert.throws(() => (evaluateXPath as any)(), 'xpathExpression must be a string'));
 
+	describe('ALL_RESULTS: the better ANY', () => {
+		it('Keeps booleans booleans', () =>
+			chai.assert.deepEqual(
+				evaluateXPath('true()', documentNode, domFacade, {}, ReturnType.ALL_RESULTS),
+				[true]
+			));
+		it('Keeps numbers numbers', () =>
+			chai.assert.deepEqual(
+				evaluateXPath('1', documentNode, domFacade, {}, ReturnType.ALL_RESULTS),
+				[1]
+			));
+		it('Keeps mixed sequences mixed', () =>
+			chai.assert.deepEqual(
+				evaluateXPath(
+					'1,true(),"test",.',
+					documentNode,
+					domFacade,
+					{},
+					ReturnType.ALL_RESULTS
+				),
+				[1, true, 'test', documentNode]
+			));
+		it('Keeps strings strings', () =>
+			chai.assert.deepEqual(
+				evaluateXPath('"string"', documentNode, domFacade, {}, ReturnType.ALL_RESULTS),
+				['string']
+			));
+		it('Keeps nodes nodes', () =>
+			chai.assert.deepEqual(
+				evaluateXPath('.', documentNode, domFacade, {}, ReturnType.ALL_RESULTS),
+				[documentNode]
+			));
+		it('Keeps attributes attributes', () => {
+			const tmp = documentNode.createElement('tmp');
+			tmp.setAttribute('class', 'value');
+			chai.assert.deepEqual(
+				evaluateXPath('@class', tmp, domFacade, {}, ReturnType.ALL_RESULTS),
+				[tmp.getAttributeNode('class')]
+			);
```

Comment:
Might be better to check that the nodes are actually the same object instead of deep equal (i.e., `chai.assert.sameMembers`)? If only because deepEqual will crash trying to traverse the cyclic structure if this somehow ends up a different instance.
(Applies to all the comparisons involving nodes)